### PR TITLE
chore(Python) : add GCD using recursion following Euclid's Algorithm

### PR DESCRIPTION
**Pull request by WORLDSAVER**
**PR Checklist:**
- [x] My submission is formatted according to the guidelines in the [contributing guide](https://github.com/MakeContributions/DSA#contribution-guidelines)
- [x] My addition is on refer on the language README.md file
- [x] My addition does not have a spelling problem
- [x] My submission has a proper and user-friendly description of the algorithm
- [x] My submission has the time complexity of the algorithm
- [x] My submission has sample input-output of the program (NOT FOR PYTHON)

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**

**Other information:**

### CODE REVIEW

**Comment by WORLDSAVER** [comment]
@Arsenic-ATG please review and allow the merge.

File: algorithms/Python/number_theory/prime_number.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#TO CHECK WHETHER A NUMBER IS PRIME OR NOT
+
+
+N = int(input())
```
Comment (Arsenic-ATG) [inline algorithms/Python/number_theory/prime_number.py:4]: Reading from stdin causes the python doctest to fail, it is preferred to provide sample tests in your program instead.

File: algorithms/Python/recursion/gcd_using_recursion.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+def gcd(a, b):
```
Comment (Arsenic-ATG) [inline algorithms/Python/recursion/gcd_using_recursion.py:1]: One PR should be preferably restricted to addition of one algorithm only.

**Review by Arsenic-ATG — CHANGES_REQUESTED** [review]
sorry for late review.

Apart from the changes mentioned in the review comments, Here are the changes that also need to be fixed before merging the PR to main

1. Update the corresponding index README.md files with en entry of each of your algorithms you added.

2. The commit message corresponding to the first program (prime_number.py) contains no info regarding addition of a whole new algorithm to the repo. ( though the commit messages are squashed before merging, it is still recommended to reword that commit message to something more fitting )

File: algorithms/Python/recursion/gcd_using_recursion.py (hunk 2 above)
Comment (WORLDSAVER) [reply algorithms/Python/recursion/gcd_using_recursion.py:1]: Ok, I'll make the changes and then commit them. Thank you for reviewing @Arsenic-ATG!

File: algorithms/Python/number_theory/prime_number.py (hunk 1 above)
Comment (WORLDSAVER) [reply algorithms/Python/number_theory/prime_number.py]: I have done the requested changes. Thanks!

File: algorithms/Python/recursion/gcd_using_recursion.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+	if b==0:
+		return a  
+	else:
+		return gcd(b, a%b)
+
```
Comment (Arsenic-ATG) [inline algorithms/Python/recursion/gcd_using_recursion.py:7]: Just add some basic texts/example calling the gcd function and pr should be good to go 👍

**Comment by WORLDSAVER** [comment]
Hi @Arsenic-ATG, I have done all the changes and also all the tests have been passed please review and allow merging.

**Review by Arsenic-ATG — APPROVED** [review]

**Comment by WORLDSAVER** [comment]
Hi @ming-tsai can you please merge this commit. It has been approved by @Arsenic-ATG . Thanks!

**Pull request merged**